c9825e6 fails on armhf

> Bus error               (core dumped)

- It builds, I see no warnings;
- A `--list=build-info` works;
- But a simple `$JTR_BIN --stdout --mask='[0-2]password[A-C]'` fails;
- arm64 **works**.

*****
https://launchpadlibrarian.net/526869316/buildlog_snap_ubuntu_bionic_armhf_john-the-ripper_BUILDING.txt.gz

```
Target CPU ................................. armv7l NEON, 32-bit LE
AES-NI support ............................. no
Target OS .................................. linux-gnueabihf
Cross compiling ............................ no
Legacy arch header ......................... arm32le.h

Optional libraries/features found:
Memory map (share/page large files) ........ yes
Fork support ............................... yes
OpenMP support ............................. yes (not for fast formats)
OpenCL support ............................. no
Generic crypt(3) format .................... yes
libgmp (PRINCE mode and faster SRP formats)  yes
128-bit integer (faster PRINCE mode) ....... no
libz (pkzip and some other formats) ........ yes
libbz2 (gpg2john extra decompression logic)  yes
libpcap (vncpcap2john and SIPdump) ......... yes
OpenMPI support (default disabled) ......... no
ZTEX USB-FPGA module 1.15y support ......... no

Install missing libraries to get any needed features that were omitted.

Configure finished.  Now "make -s clean && make -sj4" to compile.
stribog_fmt_plug.c:472:2: warning: #warning Stribog-256 and Stribog-512 formats require SSE 4.1, formats disabled [-Wcpp]
 #warning Stribog-256 and Stribog-512 formats require SSE 4.1, formats disabled
  ^~~~~~~
ar: creating aes.a
ar: creating poly1305-donna.a
ar: creating ed25519-donna.a
ar: creating secp256k1.a

Make process completed.
```

```
Version: 1.9.0-jumbo-1+bleeding-c9825e6 2021-03-08 11:40:52 +0100
Build: linux-gnueabihf 32-bit armv7l NEON AC OMP
SIMD: NEON, interleaving: MD4:2 MD5:2 SHA1:1 SHA256:1 SHA512:1
System-wide exec: /snap/john-the-ripper/current/run
System-wide home: /snap/john-the-ripper/current/run
Private home: ~/.john
OMP fallback binary: john-non-omp
$JOHN is /snap/john-the-ripper/current/run/
Format interface version: 14
Max. number of reported tunable costs: 4
Rec file version: REC4
Charset file version: CHR3
CHARSET_MIN: 1 (0x01)
CHARSET_MAX: 255 (0xff)
CHARSET_LENGTH: 24
SALT_HASH_SIZE: 1048576
SINGLE_IDX_MAX: 32768
SINGLE_BUF_MAX: 4294967295
Effective limit: Max. KPC 32768
Max. Markov mode level: 400
Max. Markov mode password length: 30
gcc version: 7.5.0
GNU libc version: 2.27 (loaded: 2.27)
Crypto library: OpenSSL
OpenSSL library version: 01010100f
OpenSSL 1.1.1  11 Sep 2018
GMP library version: 6.1.2
File locking: fcntl()
fseek(): fseeko64
ftell(): ftello64
fopen(): fopen64
memmem(): System's
times(2) sysconf(_SC_CLK_TCK) is 100
Using times(2) for timers, resolution 10 ms
HR timer: clock_gettime(), latency 1.525 us
Total physical host memory: 7975 MiB
Available physical host memory: 7883 MiB
Terminal locale string: C.UTF-8
Parsed terminal locale: UTF-8
```

--- a/src/arm32le.h
+++ b/src/arm32le.h
@@ -27,7 +27,7 @@
 #define ARCH_INT_GT_32			0
 #endif
 
-#define ARCH_ALLOWS_UNALIGNED		__ARM_FEATURE_UNALIGNED
+#define ARCH_ALLOWS_UNALIGNED		0
 #define ARCH_INDEX(x)			((unsigned int)(unsigned char)(x))
 
 #define CPU_DETECT			0

--- a/src/arm64le.h
+++ b/src/arm64le.h
@@ -28,7 +28,7 @@
 #define ARCH_INT_GT_32			0
 #endif
 
-#define ARCH_ALLOWS_UNALIGNED		__ARM_FEATURE_UNALIGNED
+#define ARCH_ALLOWS_UNALIGNED		0
 #define ARCH_INDEX(x)			((unsigned int)(unsigned char)(x))
 
 #define CPU_DETECT			0

--- a/src/opencl_rar_fmt_plug.c
+++ b/src/opencl_rar_fmt_plug.c
@@ -40,7 +40,14 @@
  *
  */
 
-#ifdef HAVE_OPENCL
+#if HAVE_OPENCL
+
+#if AC_BUILT
+#include "autoconfig.h"
+#endif
+#include "arch.h"
+
+#if ARCH_ALLOWS_UNALIGNED || __ARM_FEATURE_UNALIGNED
 
 #if FMT_EXTERNS_H
 extern struct fmt_main fmt_ocl_rar;
@@ -54,11 +61,6 @@ john_register_one(&fmt_ocl_rar);
 #include <omp.h>
 #endif
 
-#if AC_BUILT
-#include "autoconfig.h"
-#endif
-
-#include "arch.h"
 #include "sha.h"
 #include "crc32.h"
 #include "misc.h"
@@ -433,4 +435,14 @@ struct fmt_main fmt_ocl_rar = {
 
 #endif /* plugin stanza */
 
+#else
+#if !defined(FMT_EXTERNS_H) && !defined(FMT_REGISTERS_H)
+#ifdef __GNUC__
+#warning ": target system requires aligned memory access, RAR OpenCL format disabled:"
+#elif _MSC_VER
+#pragma message(": target system requires aligned memory access, RAR OpenCL format disabled:")
+#endif
+#endif
+
+#endif /* ARCH_ALLOWS_UNALIGNED || __ARM_FEATURE_UNALIGNED */
 #endif /* HAVE_OPENCL */

--- a/src/rar_fmt_plug.c
+++ b/src/rar_fmt_plug.c
@@ -45,7 +45,7 @@
 #endif
 #include "arch.h"
 
-#if ARCH_ALLOWS_UNALIGNED
+#if ARCH_ALLOWS_UNALIGNED || __ARM_FEATURE_UNALIGNED
 
 #if FMT_EXTERNS_H
 extern struct fmt_main fmt_rar;
@@ -453,4 +453,4 @@ struct fmt_main fmt_rar = {
 #endif
 #endif
 
-#endif	/* ARCH_ALLOWS_UNALIGNED */
+#endif	/* ARCH_ALLOWS_UNALIGNED || __ARM_FEATURE_UNALIGNED */

